libnavigation-core/src/main/java/com/mapbox/navigation/core/history/HistoryRecorderController.kt

```diff
@@ -0,0 +1,69 @@
+package com.mapbox.navigation.core.history
+
+import com.mapbox.base.common.logger.Logger
+import com.mapbox.base.common.logger.model.Message
+import com.mapbox.base.common.logger.model.Tag
+import com.mapbox.navigation.base.options.NavigationOptions
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigator.HistoryRecorderHandle
+
+/**
+ * Internal class that surfaces error messages and holds onto
+ * nav-native's [HistoryRecorderHandle].
+ */
+internal class HistoryRecorderController(
+    navigationOptions: NavigationOptions,
+    val logger: Logger
+) {
+    private val historyRecorderOptions = navigationOptions.historyRecorderOptions
+    private val historyFiles = HistoryFiles(navigationOptions.applicationContext, logger)
+
+    var historyRecorderHandle: HistoryRecorderHandle? = null
+        internal set
+
+    /**
+     * Used to expose a way for external developers to
+     * save history files via [MapboxNavigation.historyRecorder].
+     *
+     * This is needed to support [MapboxNavigation.resetTripSession]
+     */
+    val historyRecorder = object : MapboxHistoryRecorder {
+
+        /**
+         * @see MapboxHistoryRecorder.fileDirectory
+         */
+        override fun fileDirectory(): String? {
+            return if (historyRecorderOptions.enabled) {
+                historyFiles.absolutePath(historyRecorderOptions)
+            } else {
+                null
+            }
+        }
+
+        /**
+         * @see MapboxHistoryRecorder.saveHistory
+         */
+        override fun saveHistory(result: SaveHistoryCallback) {
+            historyRecorderHandle?.dumpHistory { filePath: String? ->
+                result.onSaved(filePath)
+            }
+            check(historyRecorderOptions.enabled) {
+                logger.e(
+                    Tag("MbxHistoryRecorderOptions"),
+                    Message("You must enable HistoryRecorderOptions in order to save history")
+                )
+            }
+            historyRecorderHandle?.apply {
+                dumpHistory { filePath: String? ->
+                    result.onSaved(filePath)
+                }
+            } ?: run {
+                logger.w(
+                    Tag("MbxHistoryRecorderOptions"),
+                    Message("dumpHistory failed - The history recorder is not initialized")
+                )
+                result.onSaved(null)
+            }
+        }
+    }
```

Comment:
```suggestion
    val historyRecorder: MapboxHistoryRecorder = MapboxHistoryRecorderImpl()

    private inner class MapboxHistoryRecorderImpl : MapboxHistoryRecorder {

        /**
         * @see MapboxHistoryRecorder.fileDirectory
         */
        override fun fileDirectory(): String? {
            return if (historyRecorderOptions.enabled) {
                historyFiles.absolutePath(historyRecorderOptions)
            } else {
                null
            }
        }

        /**
         * @see MapboxHistoryRecorder.saveHistory
         */
        override fun saveHistory(result: SaveHistoryCallback) {
            historyRecorderHandle?.dumpHistory { filePath: String? ->
                result.onSaved(filePath)
            }
            check(historyRecorderOptions.enabled) {
                logger.e(
                    Tag("MbxHistoryRecorderOptions"),
                    Message("You must enable HistoryRecorderOptions in order to save history")
                )
            }
            historyRecorderHandle?.apply {
                dumpHistory { filePath: String? ->
                    result.onSaved(filePath)
                }
            } ?: run {
                logger.w(
                    Tag("MbxHistoryRecorderOptions"),
                    Message("dumpHistory failed - The history recorder is not initialized")
                )
                result.onSaved(null)
            }
        }
    }
```
If the interface is sealed as suggested in the other comment, it cannot be implemented by an anonymous class.

Reply:
removed the inner class.

i did like a clean public interface, and wasn't too concerned with the inheritance part. taking feedback and just removed it completely in favor of a class.

thanks for ideas here, considering we're rushing for a release - i'm not opposed to ideas we have to improve this. but resolving this one